kratos/geometries/tetrahedra_3d_10.h

```diff
@@ -1031,6 +1055,27 @@ template<class TPointType> class Tetrahedra3D10 : public Geometry<TPointType>
         return shape_functions_local_gradients;
     }
 
+    /**
+     * Checks if faces are planar. We iterate for all edges and check
+     * that the sum of 0-2 and 2-1 segments is no bigger than 0-1.
+     *
+     * @return bool faces are planar or not
+     *
+     */
+    bool FacesArePlanar() const
+    {
+        constexpr double tol = 1e-6;
+        for (auto edge : this->GenerateEdges()) {
```

Comment:
```suggestion
        for (auto& edge : this->GenerateEdges()) {
```
I think otherwise you are doing a copy of a temporary object

Reply:
you are right